packages/core/src/stepper/stepper-item.tsx

```diff
@@ -0,0 +1,42 @@
+import { type ValidComponent, splitProps, JSX } from "solid-js";
+import { type ElementOf, Polymorphic, type PolymorphicProps } from "../polymorphic";
+import { useStepperContext } from "./stepper-context";
+
+export interface StepperItemOptions {
+	/** The index of this step item */
+	index: number;
+}
+
+export interface StepperItemCommonProps<T extends HTMLElement = HTMLElement> {
+	children?: JSX.Element;
+}
+
+export interface StepperItemRenderProps extends StepperItemCommonProps {
+	"data-complete"?: string;
+	"data-current"?: string;
+}
+
+export type StepperItemProps<T extends ValidComponent | HTMLElement = HTMLElement> =
+	StepperItemOptions & Partial<StepperItemCommonProps<ElementOf<T>>>;
+
+export function StepperItem<T extends ValidComponent = "div">(
+	props: PolymorphicProps<T, StepperItemProps<T>>
+) {
+	const context = useStepperContext();
+	const [local, others] = splitProps(props as StepperItemProps, ["index", "children"]);
+
+	const isCurrent = () => context.step() === local.index;
+	const isComplete = () => context.step() > local.index;
+
+	return (
+		<Polymorphic<StepperItemRenderProps>
```

Comment:
Instead of re-implementing a tab trigger it would be best to return a wrapped `Tabs.Trigger` as this will handle most of the required aria attributes.